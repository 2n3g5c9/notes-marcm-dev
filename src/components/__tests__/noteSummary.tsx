import React from 'react'
import renderer from 'react-test-renderer'

import { NoteSummary } from '@components'

jest.mock('gatsby-plugin-mdx', () => {
  return {
    // eslint-disable-next-line react/display-name
    MDXRenderer: ({ children }: { children: React.ReactNode }) => {
      return <div>{children}</div>
    },
  }
})

describe('NoteSummary', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NoteSummary />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
