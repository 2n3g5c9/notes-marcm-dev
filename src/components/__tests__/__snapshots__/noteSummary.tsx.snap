// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSummary renders correctly 1`] = `
<div>
  <div
    className="inline-block text-xs sm:text-sm"
  >
    <span
      className="font-medium inline-flex items-center px-3 py-0.5 rounded-full bg-undefined-100 text-undefined-800"
    >
      Default
    </span>
    <span
      aria-hidden="true"
      className="px-2 text-secondary"
    >
      ‚è±
    </span>
    <span
      className="text-secondary"
    >
      1 min read
    </span>
  </div>
  <a
    className="block mt-3"
    href="#"
  >
    <p
      className="font-semibold font-serif text-primary text-xl sm:text-2xl "
    >
      Default title
    </p>
    <p
      className="mt-3 text-secondary text-sm sm:text-base"
    >
      This is a default summary for a default article.
    </p>
  </a>
  <div
    className="mt-3"
  >
    <a
      className="font-semibold text-accent text-sm sm:text-base"
      href="#"
    >
      Read full note
    </a>
  </div>
</div>
`;
